### DISCUSSION TITLE
Training Model

### DISCUSSION THREAD

**Discussion by HyakuzaO** [kyleskom/NBA-Machine-Learning-Sports-Betting#404] [Q&A]
Hey guys, sorry for my rookie question but im a noob here. Just a simple question, we must train the model , get odds get data and create game than run the main script everyday for new games?

**Comment by Scandianis** [comment]
I think it is enough to train the model once a week - it takes some time and will have minimal effect on predictions if performed every day. If there are NBA matches a day, Get_Data, Get_Odds_Data, Create_Games must be executed for that day.

**Reply by Scandianis** [reply]
After training, make sure src/Predict/XGBoost_Runner.py and src/Predict/NN_Runner.py are using the recent training data.

**Reply by HyakuzaO** [reply]
How can i sure that are running the recent data?

**Reply by Scandianis** [reply]
In XGBoost_Runner.py and NN_Runner.py load_model must use the recent files:

![image](https://github.com/kyleskom/NBA-Machine-Learning-Sports-Betting/assets/23011769/19102a2e-f32d-452b-98d3-3d6c5e4a9aef)

**Comment by STRATZ-Ken** [comment]
Its important to understand what your asking.  And also important to understand what are the flaws of this project. This project is a great foundation for starting ML.  However, the amount of normal checks missing in this project are massive.  

Lets take this very basic example.  If your training set it 1000 NBA games. And of that your test set is 200 games.  

Now lets assume for some crazy reason in the last 10 days, the 50 games, away Teams just happen to win 80% of the time.  This will unbalance the test set, and if you re-train, your simply going to balance towards giving away teams an edge when its just a fluke it happened.

This is just ONE of a million examples I can come up with.  Injuries, Travel time, Travel Distance, Time zone changes, etc etc.  

If your planning to use this as is to bet, you will lose money. Just straight and simple.  If you are here to learn about machine learning, then this is an awesome project.

**Reply by Scandianis** [reply]
Having good results (65% wins) using the Under Over bets last 1 1/2 month. Bet stake adjusted with Kelly criterion in various steps. Using predictions with probability >= 75%. Also other treshold measures used.

**Reply by STRATZ-Ken** [reply]
Broken clocks are correct twice per day.  This is not my opinion, its a fact. Plenty of issues go into this in-depth.
